src/Microsoft.AspNet.Mvc.Core/Routing/AttributeRoute.cs

```diff
@@ -99,20 +100,23 @@ private InnerAttributeRoute BuildRoute(ActionDescriptorsCollection actions)
             var matchingEntries = new List<AttributeRouteMatchingEntry>();
             foreach (var routeInfo in distinctRouteInfosByGroup)
             {
+                var defaults = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase)
+                {
+                    { AttributeRouting.RouteGroupKey, routeInfo.RouteGroup },
+                };
+
                 matchingEntries.Add(new AttributeRouteMatchingEntry()
                 {
                     Order = routeInfo.Order,
                     Precedence = routeInfo.Precedence,
-                    Route = new TemplateRoute(
-                        _target,
-                        routeInfo.RouteTemplate,
-                        defaults: new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase)
-                        {
-                            { AttributeRouting.RouteGroupKey, routeInfo.RouteGroup },
-                        },
-                        constraints: null,
-                        dataTokens: null,
-                        inlineConstraintResolver: _constraintResolver),
+                    Target = _target,
+                    RouteName = routeInfo.Name,
+                    RouteTemplate = routeInfo.RouteTemplate,
+                    TemplateMatcher = new TemplateMatcher(routeInfo.ParsedTemplate, defaults),
+                    Defaults = defaults,
+                    Constraints = routeInfo.Constraints,
+                    DataTokens = EmptyRouteValueDictionary,
```

Comment:
You can just remove `DataTokens` from this class, it's never used in attribute routing.